lib/reporters/default.js

```diff
@@ -0,0 +1,252 @@
+'use strict';
+const os = require('os');
+const path = require('path');
+const stream = require('stream');
+
+const cliCursor = require('cli-cursor');
+const figures = require('figures');
+const indentString = require('indent-string');
+const ora = require('ora');
+const plur = require('plur');
+const trimOffNewlines = require('trim-off-newlines');
+const trimRight = require('trim-right');
+
+const chalk = require('../chalk').get();
+const codeExcerpt = require('../code-excerpt');
+const colors = require('./colors');
+const formatSerializedError = require('./format-serialized-error');
+const improperUsageMessages = require('./improper-usage-messages');
+const prefixTitle = require('./prefix-title');
+const whileCorked = require('./while-corked');
+
+class LineWriter extends stream.Writable {
+	constructor(dest, spinner) {
+		super();
+
+		this.dest = dest;
+		this.columns = dest.columns || 80;
+		this.spinner = spinner;
+		this.lastSpinnerText = '';
+	}
+
+	_write(chunk, encoding, callback) {
+		// Discard the current spinner output. Any lines that were meant to be
+		// preserved should be rewritten.
+		if (this.spinner) {
+			this.spinner.clear();
+		}
+
+		this._writeWithSpinner(chunk);
+		callback();
+	}
+
+	_writev(pieces, callback) {
+		// Discard the current spinner output. Any lines that were meant to be
+		// preserved should be rewritten.
+		if (this.spinner) {
+			this.spinner.clear();
+		}
+
+		const last = pieces.pop();
+		for (const piece of pieces) {
+			this.dest.write(piece.chunk);
+		}
+		this._writeWithSpinner(last.chunk);
+		callback();
+	}
+
+	_writeWithSpinner(chunk) {
+		if (!this.spinner || !this.spinner.id) {
+			this.dest.write(chunk);
+			return;
+		}
+
+		const str = chunk.toString('utf8');
+		this.lastSpinnerText = str;
+		// Ignore whitespace at the end of the chunk. We're continiously rewriting
+		// the last line through the spinner. Also be careful to remove the indent
+		// as the spinner adds its own.
+		this.spinner.text = trimRight(str).slice(2);
+		this.spinner.render();
+	}
+
+	writeLine(str) {
+		if (str) {
+			this.write(indentString(str, 2) + os.EOL);
+		} else {
+			this.write(os.EOL);
```

Comment:
No need to use `os.EOL`. Windows also supports `\n` for console output. Even `console.log` in Node.js uses `\n`: https://github.com/nodejs/node/blob/34bd9f318a164d6b2ee949793146b85b3e152d5f/lib/console.js#L141-L148